salt/grains/core.py

```diff
@@ -1372,8 +1372,7 @@ def os_data():
                 grains.pop('lsb_distrib_release', None)
             grains['osrelease'] = \
                 grains.get('lsb_distrib_release', osrelease).strip()
-        grains['oscodename'] = grains.get('lsb_distrib_codename',
-                                          oscodename).strip()
+        grains['oscodename'] = grains.get('lsb_distrib_codename', '').strip() or oscodename
```

Comment:
Because it says before "if there is no `lsb_distrib_codename`, use `oscodename`, then `.strip()` everything". Which is failing, in case `lsb_distrib_codename` contains an empty data. In this case you are happily getting an empty string, while you _already_ have a _correct_ codename from the `platform.linux_distribution`.